runtime/ts/plan/suggestion.ts

```diff
@@ -116,15 +117,16 @@ export class Suggestion {
       plan: this._planToString(this.plan),
       hash: this.hash,
       rank: this.rank,
+      relevance: this.relevance.serialize(),
       searchGroups: this.searchGroups,
       descriptionByModality: this.descriptionByModality
     };
   }
 
-  static async deserialize({plan, hash, rank, searchGroups, descriptionByModality}, arc, recipeResolver): Promise<Suggestion> {
+  static async deserialize({plan, hash, relevance, searchGroups, descriptionByModality}, arc, recipeResolver): Promise<Suggestion> {
```

Comment:
the standard pattern for serialization to/from a structure like this is to provide `toLiteral()`
and `static fromLiteral()`. If you do this then you'll find it easier to fit into the storage framework, as well as to communicate the serializable thing across the pec boundary (if necessary).

On the other hand, this pattern doesn't tend to allow extra inputs (like arc and recipeResolver are). Is there a bit more of a writeup around what the serialization is for that I can review? I'm having trouble fitting it into my overall picture of the system.

Reply:
These methods are used to read/write suggestions from store, and arc and recipeResolver are needed to parse the manifest and resolve the recipe. in similar method in Arc, the interface is 
`static async deserialize({serialization, pecFactory, slotComposer, loader, fileName, context}): Promise<Arc> `
so the similarly needed additional params are inside the first parameter object, i can do it similarly here, if that is preferred? 
should i rename the methods to toLiteral/fromLiteral? Arc has serialize/deserialize..